src/server.jl

```diff
@@ -116,20 +116,34 @@ Evaluate the code and markdown in `f` and return a vector of cells with the
 results in all available mimetypes.
 
 `output` can be a file path, or an IO stream.
+`markdown` can be used to pass an override for the file content, which is used
+    to pass the modified markdown that quarto creates after resolving shortcodes
 """
 function evaluate!(
     f::File,
     output::Union{AbstractString,IO,Nothing} = nothing;
     showprogress = true,
     options::Union{String,Dict{String,Any}} = Dict{String,Any}(),
     chunk_callback = (i, n, c) -> nothing,
+    markdown::Union{String,Nothing} = nothing,
 )
     _check_output_dst(output)
 
     options = _parsed_options(options)
     path = abspath(f.path)
     if isfile(path)
-        raw_chunks, file_frontmatter = raw_text_chunks(f)
+        display(markdown)
+        raw_chunks, file_frontmatter = if markdown === nothing
+            raw_text_chunks(f)
+        else
+            mktempdir() do dir
+                tempfile = joinpath(dir, "input.qmd")
+                open(tempfile, "w") do io
+                    write(io, markdown)
+                end
+                raw_text_chunks(tempfile)
```

Comment:
Would adding a `raw_text_chunks_str` (or some similar name) help avoiding writing to a temp file? Then have the current `raw_text_chunks` call that after reading?

Reply:
I can refactor the underlying function so it doesn't expect a file anymore

Reply:
seems that expects a path? https://github.com/MichaelHatherly/CommonMark.jl/blob/bed0be28fc8828039eb29e1c4ff65a5b3e041eb0/src/parsers/blocks.jl#L453

Reply:
https://github.com/MichaelHatherly/CommonMark.jl/blob/bed0be28fc8828039eb29e1c4ff65a5b3e041eb0/src/parsers/blocks.jl#L450, just pass a `String` to the parser object?